public/app/features/provisioning/Wizard/BootstrapStep.tsx

```diff
@@ -0,0 +1,398 @@
+      if (option.value === 'instance' && option.operation === 'pull' && settingsQuery.data?.legacyStorage) {
+        return {
+          isDisabled: true,
+          disabledReason:
+            'Pulling from repository to instance is not supported when using legacy storage. Please migrate to unified storage.',
+        };
+      }
+
+      // Disable pull instance option if there are existing dashboards or folders
+      if (option.value === 'instance' && option.operation === 'pull' && (dashboardCount > 0 || folderCount > 0)) {
+        return {
+          isDisabled: true,
+          disabledReason:
+            'Cannot pull to instance when you have existing resources. Please migrate your existing resources first.',
+        };
+      }
+
+      // Disable migrate option if there are existing files
+      if (option.operation === 'migrate' && fileCount > 0) {
+        return {
+          isDisabled: true,
+          disabledReason: 'Cannot migrate when repository has existing files. Please delete existing files first.',
+        };
+      }
+
+      if (option.value === 'instance' && otherInstanceConnected) {
+        return {
+          isDisabled: true,
+          disabledReason:
+            'Instance-wide connection is disabled because another instance is connected to this repository.',
+        };
+      }
+
+      if (option.value === 'instance' && otherFolderConnected) {
+        return {
+          isDisabled: true,
+          disabledReason: 'Instance-wide connection is disabled because folders are connected to repositories.',
+        };
+      }
+
+      return { isDisabled: false };
+    },
+    [
+      settingsQuery.data?.legacyStorage,
+      dashboardCount,
+      folderCount,
+      fileCount,
+      otherInstanceConnected,
+      otherFolderConnected,
+    ]
+  );
+
+  // Add sorted options after getOptionState is defined
+  const sortedModeOptions = useMemo(() => {
+    return [...modeOptions].sort((a, b) => {
+      const stateA = getOptionState(a);
+      const stateB = getOptionState(b);
+      return stateA.isDisabled === stateB.isDisabled ? 0 : stateA.isDisabled ? 1 : -1;
+    });
+  }, [getOptionState]);
+
+  const handleOptionSelect = useCallback(
+    (option: ModeOption) => {
+      const optionState = getOptionState(option);
+      if (optionState.isDisabled) {
+        return;
+      }
+
+      // Select the new option and update form state
+      setSelectedOption(option);
+      setValue('repository.sync.target', option.value);
+      onOptionSelect(option.operation === 'migrate');
+    },
+    [setValue, onOptionSelect, getOptionState]
+  );
+
+  const isLoading = settingsQuery.isLoading || isLoadingCounts || isLoadingFiles;
+  const hasAllData = Boolean(settingsQuery.data) && hasLoadedCounts && (!currentRepoName || filesData !== undefined);
+
+  // Store initialization function in a ref to avoid dependency issues
+  const initializeRef = useRef((options: typeof sortedModeOptions) => {
```

Comment:
I have no idea what this does and why is this necessary.

Reply:
so much fight with this one. Having a hard time to have the options enabled and the first available option selected and zero glitches.